src/inject/dynamic-theme/is-it-dark.ts

```diff
@@ -0,0 +1,71 @@
+import type {RGBA} from '../../utils/color';
+import {rgbToHSL} from '../../utils/color';
+import {tryParseColor} from './modify-css';
+import type {VariablesStore} from './variables';
+
+const IS_DARK_TRESHOLD = 0.5;
+
+let bodyObserver: MutationObserver;
+const bodyCallbacks = new Set<() => void>();
+
+async function waitTillBody(): Promise<void> {
+    return new Promise<void>((resolve) => {
+        if (!bodyObserver) {
+            resolve();
+        }
+        bodyCallbacks.add(resolve);
+    });
+}
+
+export function IsItDark(variableStore: VariablesStore, selectorText: string, styleRule: CSSStyleDeclaration): boolean | Promise<boolean> {
+    const actualFunction = () => {
+        const selectors = selectorText.split(', ');
+        if (!selectors.some((selector) => {
+            return document.documentElement.matches(selector) || document.body.matches(selector);
+        })) {
+            return false;
+        }
+
+        const backgroundValue = styleRule.getPropertyValue('background-color').trim();
+        if (!backgroundValue || backgroundValue === 'transparent') {
+            return false;
+        }
+
+        let backgroundColor: RGBA;
+        if (backgroundValue.startsWith('var(')) {
+            // Possible TODO: Check if variable isn't present and fallback to fallback value or/and
+            // subscribe to the typeChange thingy logic in variableStore.
+            backgroundColor = tryParseColor(variableStore.getRawValue(backgroundValue));
+        } else {
+            backgroundColor = tryParseColor(backgroundValue);
+        }
+
+        if (!backgroundColor) {
+            return false;
+        }
+
+        const backgroundColorHSLA = rgbToHSL(backgroundColor);
```

Comment:
This likely matters little, but I would prefer to have a better lightness algo than HSL. sRGB relative luminance (Y) would be a good choice in this case (as it's simple to calculate), although the even simpler luma (Y') will work fine too.

Reply:
Not really sure if we should switch to sRGB. We do already uses it for our image algorithm. But all modifications to CSS is being done to HSL and really would like to keep that in par. If we switch to sRGB it might be better to switch all HSL-luminance algorithms to sRGB-luminance so it will be in par and avoid weird situations.